src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -160,18 +160,20 @@ internal class CodeActionEditHandlerService : ForegroundThreadAffinitizedObject,
                 // all the changes under.
                 using (var transaction = workspace.OpenGlobalUndoTransaction(title))
                 {
-                    // ConfigureAwait(true) so we come back to the same thread as 
-                    // we do all application on the UI thread.
-                    ProcessOperations(
-                        workspace, operations, progressTracker,
-                        cancellationToken);
-
                     // link current file in the global undo transaction
+                    // Do this before processing operations, since that can change
+                    // documentIds.
                     if (fromDocument != null)
                     {
                         transaction.AddDocument(fromDocument.Id);
                     }
 
+                    // ConfigureAwait(true) so we come back to the same thread as 
```

Comment:
Comment seems stale since there's no ConfigureAwait here. (Yes, I know it's not your original doing but it's definitely wrong...)